helpers/compile/build.ts

```diff
@@ -178,48 +181,53 @@ export async function build(options: BuildOptions[]) {
  * Executes the build and rebuilds what is necessary
  * @param builds
  */
-const watch = (options: BuildOptions[]) => (result?: esbuild.BuildResult | esbuild.BuildIncremental) => {
-  if (process.env.WATCH !== 'true') return result
+const watch =
+  (allOptions: BuildOptions[]) =>
+  ([options, result]: readonly [BuildOptions, esbuild.BuildResult | esbuild.BuildIncremental]) => {
+    if (process.env.WATCH !== 'true') return result
 
-  // common chokidar options for the watchers
-  const config = { ignoreInitial: true, useFsEvents: true, ignored: ['./src/__tests__/**/*'] }
+    // common chokidar options for the watchers
+    const config = { ignoreInitial: true, useFsEvents: true, ignored: ['./src/__tests__/**/*'] }
 
-  // prepare the incremental builds watcher
-  const watched = getWatchedFiles(result)
-  const changeWatcher = createWatcher(watched, config)
+    // prepare the incremental builds watcher
+    const watched = getWatchedFiles(result)
+    const changeWatcher = createWatcher(watched, config)
 
-  // watcher for restarting a full rebuild
-  const restartWatcher = createWatcher(['./src/**/*'], config)
+    // watcher for restarting a full rebuild
+    const restartWatcher = createWatcher(['./src/**/*'], config)
 
-  // triggers quick rebuild on file change
-  changeWatcher.on('change', async () => {
-    const timeBefore = Date.now()
+    // triggers quick rebuild on file change
+    const onChange = debounce(async () => {
+      const timeBefore = Date.now()
 
-    // we handle possible rebuild exceptions
-    const rebuildResult = await handle.async(() => {
-      return result?.rebuild?.()
-    })
+      // we handle possible rebuild exceptions
+      const rebuildResult = await handle.async(() => {
+        return result?.rebuild?.()
+      })
 
-    if (rebuildResult instanceof Error) {
-      console.error(rebuildResult.message)
-    }
+      if (rebuildResult instanceof Error) {
+        console.error(rebuildResult.message)
+      }
 
-    console.log(`${Date.now() - timeBefore}ms`)
-  })
+      console.log(`${Date.now() - timeBefore}ms [${options.name ?? ''}]`)
+    }, 10)
```

Comment:
I wonder if it makes sense to have a bigger timeout (maybe not)

Reply:
Yep, I was also unsure let's experiment with it and change if needed. For now, that seems to have at least fixed the compilation issues we had in watch mode.